internal/contentcache/contentcache.go

```diff
@@ -39,29 +45,59 @@ type ContentCache struct {
 	mtimeClock timeutil.Clock
 }
 
+// RecoverFileFromCache recovers a file from the cache via metadata
+func (c *ContentCache) RecoverFileFromCache(metadataFile fs.FileInfo) {
+	debug := logger.NewDebug("recover file from cache: ")
+	// validate not a directory and matches gcsfuse pattern
+	if !metadataFile.IsDir() && matchPattern(metadataFile.Name()) {
+		var metadata gcsx.TempFileObjectMetadata
+		// TODO ezl handle '/' char by stripping it and appending ourselves
+		// so that the user can specify folder name with or without '/'
+		metadataAbsolutePath := fmt.Sprintf("%v/%v", c.tempDir, metadataFile.Name())
+		contents, err := ioutil.ReadFile(metadataAbsolutePath)
+		// TODO ezl should we exec some sort of cleanup in the cache if there are errors
+		if err != nil {
+			debug.Printf("Skip metadata file %v due to read error: %s", metadataFile.Name(), err)
+			return
+		}
+		err = json.Unmarshal(contents, &metadata)
+		if err != nil {
+			debug.Printf("Skip metadata file %v due to file corruption: %s", metadataFile.Name(), err)
+			return
+		}
+		cacheObjectKey := &CacheObjectKey{BucketName: metadata.BucketName, ObjectName: metadata.ObjectName}
+		// TODO ezl we should probably store the cached file name inside the cache file metadata instead of using the json file name
+		fileName := strings.TrimSuffix(metadataAbsolutePath, filepath.Ext(metadataFile.Name()))
+		file, err := os.Open(fileName)
```

Comment:
Some OS has a limit for open file to be 1024. So, this is not scalable. A better way to handle it is to open on demand.

Reply:
I left a TODO, will address in refactor.